proxy/proxy.go

```diff
@@ -27,62 +26,54 @@ import (
 )
 
 const restartSleep = 1 * time.Second
-const disconnectTimeout = 10 * time.Second
 
 type Proxy struct {
 	log    *zap.Logger
 	statsd *statsd.Client
 
-	config *config.Config
-
-	upstreamConfigHost string
-	localConfigHost    string
-	maxPoolSize        int
-	minPoolSize        int
-	readTimeout        time.Duration
-	writeTimeout       time.Duration
-	database           int
-	readonly           bool
+	config *config.Listener
+	lookup UpstreamManager
 
 	quit chan interface{}
 	kill chan interface{}
 
-	listeners    map[string]*listener.Listener
-	listenerLock sync.Mutex
-	listenerWg   sync.WaitGroup
+	listeners      map[string]*listener.Listener
+	listenerConfig map[string]*config.Listener
+	listenerLock   sync.Mutex
+	listenerWg     sync.WaitGroup
 
 	reservations *handlers.Reservations
 }
 
-func NewProxy(log *zap.Logger, sd *statsd.Client, config *config.Config, label, upstreamHost string, database int, minPoolSize, maxPoolSize int, readTimeout, writeTimeout time.Duration, readonly bool, maxSub, maxBlk int) (*Proxy, error) {
-	if label != "" {
-		log = log.With(zap.String("cluster", label))
+func NewProxy(ctx context.Context, listenerConfig *config.Listener, lookup UpstreamManager) (*Proxy, error) {
+	cfg := &config.Listener{}
+	*cfg = *listenerConfig
```

Comment:
What's the purpose of assigning `listenerConfig` to `cfg` this way instead of just using `cfg := listenerConfig`?

Reply:
The idea was to keep a local copy of the config. made the recommended changes of passing by value